spring-integration-xml/src/test/java/org/springframework/integration/xml/DefaultXmlPayloadConverterTests.java

```diff
@@ -143,4 +143,10 @@ public void testConvertSourceToDocument() throws Exception {
 		assertEquals("hello", childNodes.item(0).getTextContent());
 	}
 
+	@Test
+	public void testConvertBytesToDocument() throws Exception {
+		Document doc = converter.convertToDocument("<test>hello</test>".getBytes());
+		XMLAssert.assertXMLEqual(testDocument, doc);
+	}
+
```

Comment:
Needs more test coverage for the other source types, for example https://github.com/garyrussell/spring-integration/commit/fb872d696d76564224b903f80397745af069c65a but others too.